app/assets/javascripts/media_player_wrapper/mejs4_plugin_hd_toggle.es6

```diff
@@ -0,0 +1,190 @@
+'use strict';
+
+/**
+ * HD Toggle plugin
+ *
+ * A custom Avalon MediaElement 4 plugin for toggling between 2 sources
+ */
+
+// Translations (English required)
+mejs.i18n.en['mejs.hd-toggle'] = 'Toggle between HD and SD quality';
+
+// Feature configuration
+Object.assign(mejs.MepDefaults, {
+  /**
+   * @type {String}
+   */
+  hdToggleLabel: 'HD',
+  /**
+   * @type {String}
+   */
+  hdToggleText: null,
+  /**
+   * @type {String}
+   */
+  hdToggleOn: true,
+  /**
+   * @type {String}
+   */
+  hdToggleBetween: ["high", "medium"]
+});
+
+Object.assign(MediaElementPlayer.prototype, {
+  // Public variables (also documented according to JSDoc specifications)
+
+  /**
+   * Feature constructor.
+   *
+   * Always has to be prefixed with `build` and the name that will be used in MepDefaults.features list
+   * @param {MediaElementPlayer} player
+   * @param {HTMLElement} controls
+   * @param {HTMLElement} layers
+   * @param {HTMLElement} media
+   */
+  buildhdToggle(player, controls, layers, media) {
+    const t = this;
+    const hdToggleText = mejs.Utils.isString(t.options.hdToggleText) ? t.options.hdToggleText : mejs.i18n.t('mejs.hd-toggle')
+    player.qualities = [];
+
+    player.sources = $(player.domNode).find("source");
+    for (var i = 0; i < player.sources.length; i++) {
+      var src = player.sources[i];
+      for (var j = 0; j < player.options.hdToggleBetween.length; j++) {
+        if (src.getAttribute("data-quality") == player.options.hdToggleBetween[j]) {
+          player.qualities[j] = src.getAttribute("src");
+        }
+      }
+    }
+
+    player.hdtoggleButton = document.createElement('div');
+    player.hdtoggleButton.className =
+      t.options.classPrefix +
+      'button ' +
+      t.options.classPrefix +
+      'hd-toggle-button';
+    player.hdtoggleButton.innerHTML = `<button type="button" aria-controls="${ t.id }"
+      title="${ hdToggleText }">${ player.options.hdToggleLabel }</button>`;
+
+    player.hdtoggleButton.addEventListener(
+      'click',
+      t.toggleQuality.bind(t)
+    );
+
+    // Add control button to player
+    t.addControlElement(player.hdtoggleButton, 'hdToggle');
+
+    if (player.options.hdToggleOn && player.qualities[0] != null) {
+      player.hdtoggleButton.className += " mejs__hdtoggle-on";
+      player.switchStream(player.qualities[0]);
+    } else if (player.qualities[1] != null) {
+      player.switchStream(player.qualities[1]);
+    } else {
+      alert("Did not find " + hdToggleBetween[0] + " and " + hdToggleBetween[1] + " streams");
```

Comment:
Is this how we want to handle this exception case?  I know we have alerts in at least one other place but is there a better way to handle this like maybe through the flash message area.

Reply:
I agree.  If there's a pattern for handling messaging to the user, let's try to go that route.

Reply:
There's no room to do that in the embedded audio player.

Reply:
This is how it was before, but I'm ok with removing this alert altogether since it only happens when: 1. audio tags' `data-quality` attributes are misconfigured. 2. there're no audio tags.

Reply:
That's right, I keep forgetting about the embed scenario.  Would the alert show up in non-embed as well, or can we check for that and handle each environment?

Reply:
We're only using this for embed audio.